PF/PFAssignment2/HexCalc/HexCalc.java

```diff
@@ -0,0 +1,194 @@
+package GET2018.PF.PFAssignment2.HexCalc;
+
+/**
+ * This class implements various methods on Hexadecimal No
+ * 
+ * @author Mohit Sharma
+ */
+public class HexCalc {
+	String hexaDecimal;
+
+	public HexCalc(String hexaDecimal) {
+		if (hexaDecimal.equals(null)) {
+			throw new NullPointerException("HexaDecimal No is Null");
+		}
+		this.hexaDecimal = hexaDecimal;
+	}
+
+	/**
+	 * Computes Addition of two Hexadecimal No
+	 * 
+	 * @param hexaCalc
+	 * @return addition of Hexadecimal Numbers
+	 */
+	String addHexaDecimal(HexCalc hexaCalc) {
+		if (hexaCalc.equals(null)) {
```

Comment:
Above comment applicable at all places where you are calling equals method on null.